org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java

```diff
@@ -3449,6 +3452,97 @@ public static VMStandin createVMFromDefinitionFile(File eeFile, String name, Str
 		throw new CoreException(status);
 	}
 
+	/**
+	 * Returns the names of all packages provided by the given {@link IVMInstall vmInstall}, optionally restricted to the given
+	 * {@link IExecutionEnvironment execution-environment}.
+	 * <p>
+	 * The returned collection contains only distinct elements, sorted in alphabetical order.<br>
+	 * If the given VM-Install is {@link #isModularJava(IVMInstall) non modular}, the execution-environment is ignored.
+	 * </p>
+	 *
+	 * @param vm
+	 *            the vm-install to query
+	 * @param environment
+	 *            the EE to restrict the packages for, ignored for non-modular VMs
+	 * @return the distinct and alphabetically sorted collection of names of all packages in the VM
+	 * @throws CoreException
+	 *             if an error occurs or a modular VM is query for packages of Java-8 or lower
+	 * @since 3.21
+	 */
+	public Collection<String> providedVMRuntimePackages(IVMInstall vm, IExecutionEnvironment environment) throws CoreException {
+		String release = environment != null ? environment.getProfileProperties().getProperty(JavaCore.COMPILER_COMPLIANCE) : null;
+		return providedVMRuntimePackages(vm, release);
+	}
+
+	/**
+	 * Returns the names of all packages provided by the given {@link IVMInstall vmInstall}, optionally restricted to the given
+	 * {@link IExecutionEnvironment execution-environment}.
+	 * <p>
+	 * The returned collection contains only distinct elements, sorted in alphabetical order.<br>
+	 * If the given VM-Install is {@link #isModularJava(IVMInstall) non modular}, the execution-environment is ignored.
+	 * </p>
+	 *
+	 * @param vm
+	 *            the vm-install to query
+	 * @param release
+	 *            the release to restrict the packages for, ignored for non-modular VMs
+	 * @return the distinct and alphabetically sorted collection of names of all packages in the VM
+	 * @throws CoreException
+	 *             if an error occurs or a modular VM is query for packages of Java-8 or lower
+	 * @since 3.21
+	 */
+	@SuppressWarnings("restriction")
```

Comment:
use RuntimeException.getCause instead

Reply:
Ok. Done.
In general I tried to be as specific as possible so that really only the `CoreException` thrown within the `computeIfAbsent` is unwrapped.